common.go

```diff
@@ -88,7 +102,21 @@ type Range struct {
 	DataPoints []DataPoint
 }
 
+// Serialize options to args. Given that DUPLICATE_POLICY and ON_DUPLICATE depend upon the issuing command we need to specify the command for which we are generating the args for
+func (options *CreateOptions) SerializeSeriesOptions(cmd string, args []interface{}) (result []interface{}, err error) {
+	result = args
+	if options.DuplicatePolicy != "" {
+		if cmd == "TS.ADD" {
+			result = append(result, "ON_DUPLICATE", string(options.DuplicatePolicy))
+		} else {
+			result = append(result, "DUPLICATE_POLICY", string(options.DuplicatePolicy))
+		}
+	}
+	return options.Serialize(result)
+}
+
 // Serialize options to args
+// Deprecated: This function has been deprecated given that DUPLICATE_POLICY and ON_DUPLICATE depend upon the issuing command, use SerializeSeriesOptions instead
```

Comment:
@danni-m I deprecated it since we need the command name to properly generate the arguments. I'm using it internally to avoid code duplication. But the end user should be aware this is a deprecated function.